components/ThemeToggler.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <div class="theme-toggle-wrapper" @click="toggleTheme()">
+    <span>
+      <img class="theme-toggle-icon" :src="require(`~/assets/${themeImage}`)" />
+    </span>
+  </div>
+</template>
+
+<script>
+function setCookie(name, value, days) {
+  let expires = ''
+  if (days) {
+    const date = new Date()
+    date.setTime(date.getTime() + days * 24 * 60 * 60 * 1000)
+    expires = '; expires=' + date.toUTCString()
+  }
+  document.cookie = name + '=' + (value || '') + expires + '; path=/'
+}
+
+export default {
+  name: 'ThemeToggler',
+  props: {
+    theme: {
+      type: String,
+      default: 'light',
+    },
+  },
+  data() {
+    return {
+      isDarkTheme: false,
+      themeImage: this.isDarkTheme ? 'sun.png' : 'moon.png',
+    }
+  },
+  beforeMount() {
+    this.theme === 'dark'
+      ? (this.isDarkTheme = true)
+      : (this.isDarkTheme = false)
+  },
+  methods: {
+    toggleClicked(value) {
+      this.isClicked = !value
+    },
+    toggleTheme() {
+      this.isDarkTheme = !this.isDarkTheme
+      this.isDarkTheme
+        ? (this.themeImage = 'sun.png')
```

Comment:
I am observing repetition of `themeImage` and `document.documentElement.className` could we look into way to reduce this